qiskit_nature/circuit/library/ansaetze/evolved_operator_ansatz.py

```diff
@@ -0,0 +1,149 @@
+
+
+class EvolvedOperatorAnsatz(BlueprintCircuit):
+    """The evolved operator ansatz."""
+
+    def __init__(self,
+                 operators: Optional[Union[OperatorBase, List[OperatorBase]]] = None,
+                 reps: int = 1,
+                 evolution: Optional[EvolutionBase] = None,
+                 insert_barriers: bool = False,
+                 name: str = 'EvolvedOps') -> None:
+        """
+        Args:
+            operators: The operators to evolve.
+            reps: The number of times to repeat the evolved operators.
+            evolution: An opflow converter object to construct the evolution.
+                Defaults to Trotterization.
+            insert_barriers: Whether to insert barriers in between each evolution.
+            name: The name of the circuit.
+        """
+        if evolution is None:
+            evolution = PauliTrotterEvolution()
+
+        super().__init__(name=name)
+        self._operators = None
+        self._evolution = evolution
+        self._reps = reps
+        self._insert_barriers = insert_barriers
+
+        # use setter to set operators
+        self.operators = operators
+
+    def _check_configuration(self, raise_on_failure: bool = True) -> bool:
+        if self.operators is None:
+            if raise_on_failure:
+                raise ValueError('The operators are not set.')
+            return False
+
+        if self.reps < 0:
+            if raise_on_failure:
+                raise ValueError('The reps cannot be smaller than 0.')
+            return False
+
+        return True
+
+    @property
+    def reps(self) -> int:
+        """The number of times the evolved operators are repeated."""
+        return self._reps
+
+    @reps.setter
+    def reps(self, r: int) -> None:
+        """Sets the number of times the evolved operators are repeated."""
+        self._invalidate()
+        self._reps = r
+
+    @property
+    def evolution(self) -> EvolutionBase:
+        """The evolution converter used to compute the evolution."""
+        return self._evolution
+
+    @evolution.setter
+    def evolution(self, evol: EvolutionBase):
+        """Sets the evolution converter used to compute the evolution."""
+        self._invalidate()
+        self._evolution = evol
+
+    @property
+    def operators(self) -> List[OperatorBase]:
+        """The operators that are evolved in this circuit."""
+        return self._operators
+
+    @operators.setter
+    def operators(self, operators: Union[OperatorBase, List[OperatorBase]]) -> None:
+        """Set the operators to be evolved."""
+        if not isinstance(operators, list):
+            operators = [operators]
+
+        if len(operators) > 1:
+            num_qubits = operators[0].num_qubits
+            if any(operators[i].num_qubits != num_qubits for i in range(1, len(operators))):
+                raise ValueError('All operators must act on the same number of qubits (for now).')
```

Comment:
```suggestion
                raise ValueError('All operators must act on the same number of qubits.')
```
I would remove the `(for now)` as that seems to imply we will change. We may well change this going forwards but I think its best left out of the message